akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -931,12 +931,24 @@ private[akka] class ShardRegion(
         val coordinatorMessage =
           if (cluster.state.unreachable(membersByAge.head)) s"Coordinator [${membersByAge.head}] is unreachable."
           else s"Coordinator [${membersByAge.head}] is reachable."
-        log.warning(
-          "{}: Trying to register to coordinator at [{}], but no acknowledgement. Total [{}] buffered messages. [{}]",
-          typeName,
-          actorSelections.mkString(", "),
-          shardBuffers.totalSize,
-          coordinatorMessage)
+        val bufferSize = shardBuffers.totalSize
+        if (bufferSize > 0) {
+          if (log.isWarningEnabled) {
+            log.warning(
+              "{}: Trying to register to coordinator at [{}], but no acknowledgement. Total [{}] buffered messages. [{}]",
+              typeName,
+              actorSelections.mkString(", "),
+              bufferSize,
+              coordinatorMessage)
+          } else {
+            log.debug(
+              "{}: Trying to register to coordinator at [{}], but no acknowledgement. No buffered messages yet. [{}]",
+              typeName,
+              actorSelections.mkString(", "),
+              coordinatorMessage)
+          }
+        }
```

Comment:
There is something wrong here. 

This is only logged when `bufferSize > 0`, but the debug level logs says: "No buffered messages yet".

Reply:
Doh, else is attached to the wrong if